pkg/languagedetection/util/annotations.go

```diff
@@ -6,13 +6,17 @@
 // Package util provides util type definitions and helper methods for the language detection client and handler
 package util
 
+import "regexp"
+
 const (
 
 	// AnnotationPrefix represents a prefix of the language detection annotations
-	AnnotationPrefix string = "apm.datadoghq.com/"
+	AnnotationPrefix string = "internal.dd.datadoghq.com/"
 )
 
+var AnnotationRegex = regexp.MustCompile(`internal\.dd\.datadoghq\.com\/(init)?\.?(.+?)\.detected_langs`)
```

Comment:
Defined the regex here and allowed exporting it in order to avoid defining the same regex again in the `kubeapiserver` collector for deployments. It can be reused.

Reply:
Might we run into issues if the container is not an init container but starts with the prefix `init`? e.g. regexr.com/7oisl

Maybe we could change it to
```
internal\.dd\.datadoghq\.com\/(init\.)?(.+?)\.detected_langs
```
which requires the prefix be `init.`, and you don't use the value of the group anyway

Reply:
Great catch!!

This has been there for a while.

I fixed it.  

This, however, doesn't handle the case when the user names the container with `init.`. But IMHO this is an extreme edge case that we don't need to handle.